python/lsst/ip/isr/deferredCharge.py

```diff
@@ -567,11 +571,19 @@ def run(self, exposure, ctiCalib, gains=None):
         # true, but no external gains were supplied, use the nominal
         # gains listed in the detector.  Finally, if useGains is
         # false, fake a dictionary of unit gains for ``gainContext``.
-        if self.config.useGains:
+        useGains = True
+        if "USEGAINS" in ctiCalib.getMetadata().keys():
+            useGains = ctiCalib.getMetadata()["USEGAINS"]
+            self.log.info(f"useGains = {useGains} due to calibration")
+        else:
+            useGains = self.config.useGains
+            self.log.info(f"useGains = {useGains} due to config")
```

Comment:
I think this log could give more info that `USEGAINS was not found in calibration metadata` or something.